falcon/request.py

```diff
@@ -524,6 +534,12 @@ def cookies(self):
         return self._cookies.copy()
 
     @property
+    def json(self):
```

Comment:
Consider not treat `.json` as property but as method.

The `python-requests` did the same thing before but [they changed it to method ultimately](https://github.com/kennethreitz/requests/commit/1451ba0c6d395c41f86da35036fa361c3a41bc90). I quote the reason here:

> A very specific one. It wasn't good design in the first place. Attributes shouldn't through errors on decoding JSON, but not throwing an error would be misleading and might make a user think there was no JSON. As a method, it is perfectly legal for it through an exception. And that's how it should be.

It's reasonable to me. And now we can avoid to repeat the same mistake they did before.